Merge upstream

gheskett opened the pull request:
May as well I guess

hensldm committed:
Add ifndef _FINALROM to kmcprintf to avoid building when not debug version (#93)

mkst committed:
Add wget as requirement in README (#98)

someone2639 commented on README.md:
````diff
@@ -32,12 +32,13 @@ The build process requires the following packages:
 ```bash
 sudo apt update
-sudo apt install build-essential python3
+sudo apt install build-essential python3 wget
````
We don't use wget do we? i thought that was just to download the versions of gcc for matching

gcsmith committed:
fetch correct tools when building on macOS (#100)

kdehl16-web commented:
re-run this

AngheloAlf committed:
Remove usage of `pull_request_target` and GHA secrets from CI (#105)

* Remove usage of `pull_request_target` and GHA secrets from CI

Uses the same docker approach from the GC/Wii community.

There's a write up about this approach here: https://github.com/AngheloAlf/drmario64/pull/19

The main motivation for getting rid of `pull_request_target` is that it was making this repo a target for bad actors trying to steal repository secrets.

`pull_request_target` bypasses the require approval before running CI on a PR check, making it an easy target for bad people.

This approach uses a docker image that contains the important and unique dependencies needed to build the repo

* whoops
